core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/DefaultEncodingFactory.java

```diff
@@ -325,32 +325,42 @@ static ColumnPageCodec selectCodecByAlgorithmForFloating(SimpleStatsResult stats
     //Here we should use the Max abs as max to getDatatype, let's say -1 and -10000000, -1 is max,
     //but we can't use -1 to getDatatype, we should use -10000000.
     double absMaxValue = Math.max(Math.abs(maxValue), Math.abs(minValue));
-    if (decimalCount == 0) {
+    if (decimalCount == 0 && srcDataType == DataTypes.FLOAT) {
```

Comment:
no need to check for decimalCount == 0. If dataType is FLOAT call  getColumnPageCodec(..)